libraries/c_sdk/standard/https/include/iot_https_client.h

```diff
@@ -0,0 +1,890 @@
+ * amount of time. Many webservers may close the connection after 30-60 seconds. The state of pConnHandle will still be
+ * in a connected state if this happens. If the server closed the connection, then the next request on the connection
+ * will fail to send with a network error and the connection will move to a closed state.
+ *
+ * Also keep in mind that some HTTP servers do not accept persistent requests. Some HTTP servers will ignore that the
+ * request contains the "Connection: keep-alive" header and close the connection immediately after sending the response.
+ * If this happens, then the next request on the connection will fail to send with a network error and the connection
+ * will close.
+ *
+ * To know if the connection was closed by the server, debug logging can be turned on to view the network error code
+ * received. Debug logging is configured when @ref IOT_LOG_LEVEL_HTTPS is set to @ref IOT_LOG_DEBUG in iot_config.h.
+ *
+ * #IotHttpsConnectionInfo_t.userBuffer is used to store the internal context and therefore, multiple threads
+ * calling this function simultaneously must ensure to use different #IotHttpsConnectionInfo_t objects.
+ *
+ * See @ref connectionUserBufferMinimumSize for information about the user buffer configured in
+ * #IotHttpsConnectionInfo_t.userBuffer needed to create a valid connection handle.
+ *
+ * @param[out] pConnHandle - Handle returned representing the open connection. NULL if the function failed.
+ * @param[in] pConnInfo - Configurations for the HTTPS connection.
+ *
+ * @return One of the following:
+ * - #IOT_HTTPS_OK if the connection was successful.
+ * - #IOT_HTTPS_CONNECTION_ERROR if the connection failed.
+ * - #IOT_HTTPS_INVALID_PARAMETER if NULL parameters were passed in.
+ * - #IOT_HTTPS_INTERNAL_ERROR if there was an error creating resources for the connection context.
+ *
+ * <b>Example</b>
+ * @code{c}
+ * // An initialized network interface.
+ * IotNetworkInterface_t* pNetworkInterface;
+ *
+ * // Parameters to HTTPS Client connect.
+ * IotHttpsConnectionInfo_t connInfo = IOT_HTTPS_CONNECTION_INFO_INITIALIZER;
+ * IotHttpsConnectionHandle_t connHandle = IOT_HTTPS_CONNECTION_HANDLE_INITIALIZER;
+ * uint8_t* pConnUserBuffer = (uint8_t*)malloc(connectionUserBufferMinimumSize);
+ *
+ * // Set the connection configuration information.
+ * connInfo.pAddress = "www.amazon.com";
+ * connInfo.addressLen = strlen("www.amazon.com");
+ * connInfo.port = 443;
+ * connInfo.flags = 0;
+ * connInfo.pAlpnProtocols = "alpnproto0,alpnproto1"
+ * connInfo.pCaCert = HTTPS_TRUSTED_ROOT_CA; // defined elsewhere
+ * connInfo.caCertLen = sizeof( HTTPS_TRUSTED_ROOT_CA );
+ * connInfo.userBuffer.pBuffer = pConnUserBuffer;
+ * connInfo.userBuffer.bufferLen = connectionUserBufferMinimumSize;
+ * connInfo.pClientCert = TLS_CLIENT_CERT;
+ * connInfo.clientCertLen = sizeof( TLS_CLIENT_CERT );
+ * connInfo.pPrivateKey = TLS_CLIENT_PRIV_KEY;
```

Comment:
These values are not used. The private key and client certificate provisioned at the start of the application are used by the network layer. These parameters are here because these parameters still exist in the network interface. In the future we are going to want to allow the user to customize their key into the application while at the same time still allowing usage of secure element keys. But how these parameters look and are passed into applications still needs to be decided upon.